ADR/partner-tasks.md

```diff
@@ -0,0 +1,34 @@
+# Partner Tasks in Build/Test Pipelines
+
+Date: 2023-03-06
+
+## Status
+
+Provisional
+
+## Context
+
+* As a Red Hat Partner, I would like to offer our service's capability as a Tekton Task that would be executed in a user's build/test Pipeline on StoneSoup.
+* As a StoneSoup maintainer, I would like to provide a way to ensure the above Tekton Task is validated and approved for use in the build/test Pipeline on StoneSoup.
+
+Associated business ask
+https://issues.redhat.com/browse/STONE-549
+
+## Decision
+
+### Plumbing
+
+1. Setup a new Git repository.
```

Comment:
One repo for all partner tasks from all partners?

Or, one repo per partner per task?

Or, one repo per partner?

Reply:
One repo for all partners to begin with.

Reply:
I left it for implementation - but it should be in a new Github organization.

Reply:
OK - does the TKNECO team own this new repo, or does the STONEBLD team own this new repo?

Reply:
STONEBLD team would own this repo.

Reply:
Cannot we just reuse build-definitions and create there folder partner-tasks? Especially when the tasks should be moved to build-definitions anyway, this way we will have just one source.

Reply:
@ralphbean in TKNECO, the current idea is that we'll have "on central repo" (with almost all task, from RH teams, from partners, …) that would be "generated"/populated from any number of smaller repositories — this would allow any teams to work with the tools they want, and "release" their task when they think it's ready, so that it gets "automatically pulled" in the "Red Hat Tekton Resource Catalog" 👼🏼

Reply:
What @Michkov suggests would be dead-simple for the short term, and avoid engineering some new repo with a new CI propagation pipeline when we know we want to move the partner content somewhere else in the medium/long-term.

Reply:
I'm generally OK with that, except that, we'd have to have non-red hatters be part of the same repository/org. More of a stylistic concern for me. Not a strong opinion.

Reply:
Added to "Alternatives".

Reply:
If users are made part of the organization, we would have to reassess our assumptions about security (i.e. teams being private is enough to hide them). Would it not be enough to have partners just submit forks? Do individuals need to be members of the organization in order to fully approve and merge PRs?

Reply:
https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1678804269686459 See thread on the concern.